server/src/controllers/authMiddleware.ts

```diff
@@ -0,0 +1,46 @@
+import { Next, Request, Response } from "restify";
+import * as jwt from "jsonwebtoken";
+import { UserRepository } from "../repositories/user_repository";
+import { Role } from "../entities/role";
+import { container } from "../registry";
+
+type CallbackMiddleware = (req: Request, res: Response, next: Next) => void;
```

Comment:
This is not really a callback function it's just a middleware function

Reply:
Also your type error from the tests might be from this. should return a promise of void